### ISSUE TITLE
how to get the figure.7？

### ISSUE THREAD

**Issue by GuangtaoLyu** [XIAO4579/Vlm-interpretability#2]
Dear Authors,

Thank you for your excellent work and valuable contributions to the field. Your paper has been incredibly insightful.

I'm trying to reproduce Figure 7 ("Feature encodings across layers for different images") using the provided `tsne.py` as a reference, but I'm encountering some discrepancies. 

In my results, the feature encodings from the first(yellow ) and last layers(purple ) appear quite close, while those from layers 19-31 show more divergence. Additionally, the yellow and purple clusters (representing different image categories) consistently appear closer in my plots than in your published figure, which seems to contradict the conclusions in the paper.

Could you please share the specific code used to generate Figure 7? 

Thank you for your time and assistance. I look forward to your response.

<img width="5700" height="4740" alt="Image" src="https://github.com/user-attachments/assets/bf8d6712-2553-48ce-a928-7dfb682dfa60" />
"What is the color of the flower? Answer the question with a single word."
"What is the main object of the image? Answer the question with a single word."

<img width="5700" height="4740" alt="Image" src="https://github.com/user-attachments/assets/3a37f2bd-8687-4a24-8f9d-2cec5610a541" />

**Comment by XIAO4579** [comment]
1) Thanks for reading the paper and for attempting to reproduce Fig. 7 — we appreciate the careful look and the thoughtful issue.

2) On interpretation: t-SNE reliably preserves local neighborhoods but not global geometry. That means cluster-to-cluster distances, the overall “ordering” of groups, or whether one color looks like it’s at the “start” or “end” of a chain in 2D are not stable indicators. Seeing the first (yellow) and last (purple) layers appear close in 2D while mid layers look more spread does not by itself contradict our conclusions; what matters more is neighborhood overlap rather than absolute 2D distances.

3) On hyperparameters: global layouts in t-SNE are very sensitive to settings and seeds. Please try varying perplexity (e.g., 5, 10, 30, 50), learning rate (e.g., 200–1000), early exaggeration (e.g., 4–12), and random_state. Using init="pca" and metric="cosine" for feature vectors often improves stability. Run several seeds and focus on whether nearest-neighbor assignments are consistent, not just the visual arrangement.

**Comment by GuangtaoLyu** [comment]
> 1. Thanks for reading the paper and for attempting to reproduce Fig. 7 — we appreciate the careful look and the thoughtful issue.
> 2. On interpretation: t-SNE reliably preserves local neighborhoods but not global geometry. That means cluster-to-cluster distances, the overall “ordering” of groups, or whether one color looks like it’s at the “start” or “end” of a chain in 2D are not stable indicators. Seeing the first (yellow) and last (purple) layers appear close in 2D while mid layers look more spread does not by itself contradict our conclusions; what matters more is neighborhood overlap rather than absolute 2D distances.
> 3. On hyperparameters: global layouts in t-SNE are very sensitive to settings and seeds. Please try varying perplexity (e.g., 5, 10, 30, 50), learning rate (e.g., 200–1000), early exaggeration (e.g., 4–12), and random_state. Using init="pca" and metric="cosine" for feature vectors often improves stability. Run several seeds and focus on whether nearest-neighbor assignments are consistent, not just the visual arrangement.

Thank you for your timely and careful reply.
